### ISSUE TITLE
False Negative UnnecessaryApply

### ISSUE THREAD

**Issue by scottkennedy** [arturbosch/detekt#1485]
## Expected Behavior
UnnecessaryApply is reported.

## Observed Behavior
UnnecessaryApply is not reported

## Steps to Reproduce
```kt
firebaseHttpMetric?.apply {
    setHttpResponseCode(response.code())
}
```

## Context
This code is better written as `firebaseHttpMetric?.setHttpResponseCode(response.code())`

## Your Environment
* Version of detekt used: RC13
* Version of Gradle used (if applicable): 5.2.1
* Operating System and version: Various

**Comment by arturbosch** [comment]
What's a case we can't decide now due to no type resolution.
`?.` can mean that the apply part should only be executed when the target is not null.
For example:
```
firebaseHttpMetric?.apply {
    logInfo { "bla" }
}
```
where `logInfo` is not art of `firebaseHttpMetric`. There is a test case and note about this in the Spec.

**Comment by scottkennedy** [comment]
If `logInfo` is not part of `firebaseHttpMetric`, then I'd say it's still an unnecessary apply.

I think that would look much better as:

```kt
if (firebaseHttpMetric != null) {
    logInfo { "bla" }
}
```

`apply` suggests you're applying something to the object, which isn't the case there, and it certainly makes the code harder to understand.

**Comment by arturbosch** [comment]
Good explanation, I'm convinced.
